### PR TITLE
Chain Communities Semantics – Server Controllers

### PR DESCRIPTION
**Pull request by rbennettcw**
## Link to Issue
Closes: #5317 (server controllers refactor)
Closes: #5356 (groups table)
Closes: #5357 (polls table)

## Description of Changes
Replaces "chain" with "community" for the following:
- Server controllers and related routes
- Error messages
- `checkBan` and `validateOwner` functions
- Groups model/table (migration)
- Polls model/table (migration)
- `validateChain` and `validateAuthor` middleware
- Relevant frontend axios calls

## Test Plan
- Existing unit/API tests updated
- CA testing– lots of changes, so this is the succinct list of features to test:
  -  Run migration
  - CRUD communities
  - CRUD threads
  - CRUD comments
  - Create/Delete reaction on threads and comments
  - Create poll, vote, delete poll
  - Search threads, comments, profiles, communities
  - CRUD topics
  - View proposal
  - Login with magic
  - Discord bot
  - CRUD groups (optional)

## Deployment Plan
- Run migration

## Other Considerations
N/A

### CODE REVIEW

**Review by CowMuon — APPROVED** [review]
low key thrilling to see 93 files updated for a 1 pointer.

**Review by timolegros — CHANGES_REQUESTED** [review]
`create_community.ts` contains references to chain that haven't been updated:
- `CreateCommunityResult.chain`
- `existingBaseChain`
- `oldChain`
- Errors
- returns `{ chain: chain.toJSON, ... }

Unlike model changes, variable changes don't need to be so large (and therefore hard to correctly review).  For future server-side PRs please try to keep them as small as possible so it is very easy to review e.g. a few routes per PR. This is especially true because review for these PRs involves inspecting `chain` strings in every edited/dependent file. I'd rather review 3-4 PRs with well-defined scopes than 1 large one.

I've reviewed about half the files but I need to step out so posting this review as is for now. Will review the other half later tonight.

File: packages/commonwealth/client/scripts/controllers/server/polls.ts [hunk 1]
```diff
@@ -46,12 +46,12 @@ class PollsController {
...
   }
 
-  public async fetchPolls(chainId: string, threadId: number) {
+  public async fetchPolls(communityId: string, threadId: number) {
     await $.ajax({
       url: `${app.serverUrl()}/threads/${threadId}/polls`,
       type: 'GET',
       data: {
-        chain: chainId,
+        chain: communityId,
```
[the remark this answers is not in the document]
Comment (rbennettcw) [reply packages/commonwealth/client/scripts/controllers/server/polls.ts:54]: Sure that makes sense.

File: packages/commonwealth/server/controllers/server_communities_methods/delete_community.ts [hunk 2]
```diff
@@ -34,13 +34,13 @@ export async function __deleteCommunity(
     throw new AppError(Errors.NeedChainId);
```
[the remark this answers is not in the document]
Comment (rbennettcw) [reply packages/commonwealth/server/controllers/server_communities_methods/delete_community.ts:34]: Yeah, we might as well go all the way

**Review by timolegros — CHANGES_REQUESTED** [review]
- Missing all updates in `server_proposals_controller.ts`.
- Missing updates in `update_community.ts` (mainly `chain` variable that is referenced throughout).

This completes the second half of my review. Focused mostly on updates that were missing. Once the requested changes are implemented I will re-review and focus only on the included changes which I suspect won't result in many (if any) updates.

**Comment by rbennettcw** [comment]
So many rippling changes 😅 – This is nearly done, I'm currently:
- Updating response payloads to return community instead of chain, where applicable
- Updating the chain validation middleware to support both `chain` and `community` param variations so we can transition
- Updating the frontend axios calls accordingly

It's still scoped to the same routes, but more thorough.

**Comment by lzach83** [comment]
I've tested everything I knew how and followed the test plan. Looks great and can't find anything that's broken.

**Review by timolegros — CHANGES_REQUESTED** [review]
- redirection is broken when creating an ERC20 or ERC721 community. It goes to `http://localhost:8080/undefined` even though the community was successfully created.
- ~~Getting 400 error when trying to update a topic~~ -> this issue is also on master: #5593
- Can't select topics to connect in Discordbot settings on manage community page

**Review by timolegros — APPROVED** [review]

**Pull request merged**